pom.xml

```diff
@@ -42,6 +42,12 @@
         </pluginRepository>
     </pluginRepositories>
 
+    <properties>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <maven.compiler.target>11</maven.compiler.target>
+        <maven.compiler.source>11</maven.compiler.source>
```

Comment:
Is it required to explicitly define the `target` and `source`? Doesn't it pull this in from whatever parent POM it uses? 

The current state of the plugin can't be used under Openfire 4.8.0 or later (because of the MINA references). That means that it now can _only_ run on versions of Openfire that support Java 8. It feels strange to then compile this for 11.

Whenever we upgrade the code to be compliant with Openfire 4.8 and later, we need to update the parent POM to that of a later Openfire version. That automatically gets us the minimum restriction set to Java 11. That probably means that there's no need to explicitly define that in this plugin either.

Reply:
I don't think you got my point. This project defines a 'parent pom' (which is a reference to Openfire). In this parent pom, there already is a definition of target and source. Therefor, the 'effective pom' of this plugin already defines a target and source.

Your changes are now overriding that. As the target and source in the parent pom are even more closely aligned with the version of Openfire that the plugin is aligned with, you probably should not override that at all, except for in very specific circumstances.